### PR TITLE
Update testing apparatus to use F4 v4.6.0

### PR DESCRIPTION
**Pull request by acoburn**
Resolves: https://jira.duraspace.org/browse/FCREPO-2147

### CODE REVIEW

File: src/test/java/org/fcrepo/camel/integration/FcrepoDeleteIT.java [hunk 1]
```diff
@@ -182,15 +185,15 @@ public void testDeleteNestedContainer() throws InterruptedException {
...
-            Assert.assertTrue(exchange.getIn().getBody(String.class).contains("Not Found"));
-        }
+        goneEndpoint.getExchanges().forEach(exchange -> {
+            assertTrue(exchange.getIn().getHeader(Exchange.CONTENT_TYPE, String.class).contains("application/rdf+xml"));
+            assertTrue(exchange.getIn().getBody(String.class).contains("Discovered tombstone"));
+        });
+
+        notFoundEndpoint.getExchanges().forEach(exchange -> {
+            assertTrue(exchange.getIn().getHeader(Exchange.CONTENT_TYPE, String.class).contains("text/html"));
+            assertTrue(exchange.getIn().getBody(String.class).contains("Not Found"));
```
Comment (ajs6f) [inline src/test/java/org/fcrepo/camel/integration/FcrepoDeleteIT.java]: This seems a bit rickety. Are the HTTP response codes not available for this sort of thing?

Comment (acoburn) [inline src/test/java/org/fcrepo/camel/integration/FcrepoDeleteIT.java]: Using response codes would be much better; here I was just making it work with 4.6.0

Comment (ajs6f) [inline src/test/java/org/fcrepo/camel/integration/FcrepoDeleteIT.java]: Okay, not trying to slow anything down. Let's just set a flag to switch to the more type-safe approach when there is a chance.

Comment (acoburn) [inline src/test/java/org/fcrepo/camel/integration/FcrepoDeleteIT.java]: actually, we're already checking the response codes, e.g.: https://github.com/fcrepo4-exts/fcrepo-camel/blob/master/src/test/java/org/fcrepo/camel/integration/FcrepoBinaryGetIT.java#L90

Comment (acoburn) [inline src/test/java/org/fcrepo/camel/integration/FcrepoDeleteIT.java]: i.e. checking the response body is just a further check and (arguably) not needed.

Comment (ajs6f) [inline src/test/java/org/fcrepo/camel/integration/FcrepoDeleteIT.java]: Well, it's up to you. I think checking for a message as specific as `"Discovered tombstone"` seems pretty brittle.

Comment (acoburn) [inline src/test/java/org/fcrepo/camel/integration/FcrepoDeleteIT.java]: I don't disagree. What I'd suggest is leaving it as it is for now and just removing those blocks the next time it changes in Fedora.

Comment (ajs6f) [inline src/test/java/org/fcrepo/camel/integration/FcrepoDeleteIT.java]: Okay by me, although I don't quite why we'd wait for the next release... but it shouldn't make any diff now and we don't want to slow things down even more.

Comment (acoburn) [inline src/test/java/org/fcrepo/camel/integration/FcrepoDeleteIT.java]: What I'm saying is this: I'm not going to remove all of those blocks right now. We can either wait until next week (or later) when I _could_ remove those blocks, and have this PR sit until then, or we can remove them as some future, unspecified point. Either way, I'm not removing them today. If anyone wants to make a PR on PR to remove them, I'd be supportive of that.

Comment (ajs6f) [inline src/test/java/org/fcrepo/camel/integration/FcrepoDeleteIT.java]: I'm not trying to stop you-- I just don't want to lose track of this. How about sticking a `TODO` in these?

Comment (acoburn) [inline src/test/java/org/fcrepo/camel/integration/FcrepoDeleteIT.java]: ok, well if it's a question of adding a TODO or removing a line, I'd rather remove the line. Let's just let this sit until I can return to it.

**Pull request merged**